contrib/compare_ids_test.bzl

```diff
@@ -24,19 +24,15 @@ def _compare_ids_test_impl(ctx):
     if (len(tar_files) == 1 and not ctx.attr.id):
         fail("One tar provided. Need either second tar or an id to compare it to.")
 
-    tars_string = ""
-    for tar in tar_files:
-        tars_string += tar.short_path + " "
-
     runfiles = ctx.runfiles(files = tar_files + [ctx.file._id_extract_script])
 
     ctx.actions.expand_template(
         template = ctx.file._executable_template,
         output = ctx.outputs.executable,
         substitutions = {
             "{id}": ctx.attr.id,
-            "{tars}": tars_string,
-            "{id_script_path}": ctx.file._id_extract_script.short_path,
+            "{tars}": repr([i.short_path for i in tar_files]),
```

Comment:
These can be passed as args to a pythons script, no need to use a template.
You can use ctx.actions.run (e.g., https://github.com/bazelbuild/rules_docker/blob/master/contrib/passwd.bzl#L80) using as executable the py_binary target (note how the example uses the //container:build_tar target as the executable)